sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFilters.scala

```diff
@@ -73,9 +74,11 @@ private[orc] object OrcFilters extends Logging {
     if (HiveUtils.isHive23) {
       DatasourceOrcFilters.createFilter(schema, filters).asInstanceOf[Option[SearchArgument]]
     } else {
-      val dataTypeMap = schema.map(f => f.name -> f.dataType).toMap
+      val dataTypeMap = schema.map(f => quoteIfNeeded(f.name) -> f.dataType).toMap
+      // TODO (SPARK-25557): ORC doesn't support nested predicate pushdown, so they are removed.
```

Comment:
does ORC support column name with dot?

Reply:
ORC already supports column name with `dots`. In fact, ORC API is using the same quotation method I purposed in this PR, but we add the quotation if the column name contains dot right before calling ORC filter APIs.